tests/unit/views/notifications/test_project_crew_notification.py

```diff
@@ -0,0 +1,400 @@
+    db_session.add(ridcully_member)
+    db_session.commit()
+    return ridcully_member
+
+
+@then(
+    parsers.parse(
+        "{user} gets notified with {notification_string} about the invitation"
+    )
+)
+@then(
+    parsers.parse("{user} gets notified with {notification_string} about the addition")
+)
+@then(
+    parsers.parse(
+        "{user} gets notified with {notification_string} about the acceptance"
+    )
+)
+@then(parsers.parse("{user} gets notified with {notification_string} about the change"))
+def then_user_notification(
+    user,
+    notification_string,
+    user_vimes,
+    user_ridcully,
+    user_vetinari,
+    project_expo2010,
+    vimes_promoter,
+    ridcully_member,
+) -> None:
+    user_dict = {
+        "Ridcully": user_ridcully,
+        "Vimes": user_vimes,
+        "Vetinari": user_vetinari,
+    }
+    preview = models.PreviewNotification(
+        models.ProjectCrewMembershipNotification,
+        document=ridcully_member.project,
+        fragment=ridcully_member,
+    )
+    user_notification = models.NotificationFor(preview, user_dict[user])
+    view = user_notification.views.render
+    assert (
+        view.activity_template().format(
+            actor=ridcully_member.granted_by.fullname,
+            project=ridcully_member.project.joined_title,
+            user=ridcully_member.user.fullname,
+        )
+        == notification_string
+    )
+
+
+@given(
+    parsers.parse(
+        "Vetinari invited Ridcully with role {role} to the Ankh-Morpork 2010 project"
+    ),
+    target_fixture='ridcully_member',
+)
+@when(
+    parsers.parse(
+        "Vetinari invites Ridcully with role {role} to the Ankh-Morpork 2010 project"
+    ),
+    target_fixture='ridcully_member',
+)
+def when_invite_ridcully_member(
+    role,
+    db_session,
+    client,
+    login,
+    user_vimes,
+    user_ridcully,
+    project_expo2010,
+    user_vetinari,
+) -> models.ProjectCrewMembership:
+    roles = [_r.strip() for _r in role.split(',')]
+    is_editor = 'editor' in roles
+    is_promoter = 'promoter' in roles
+    is_usher = 'usher' in roles
+    ridcully_member = models.ProjectCrewMembership(
+        parent=project_expo2010,
+        user=user_ridcully,
+        is_editor=is_editor,
+        is_promoter=is_promoter,
+        is_usher=is_usher,
+        granted_by=user_vetinari,
+    )
+    ridcully_member.record_type = MEMBERSHIP_RECORD_TYPE.INVITE
+    db_session.add(ridcully_member)
+    db_session.commit()
+    return ridcully_member
+
+
+@when(
+    "Ridcully accepts the invitation to be a crew member of the Ankh-Morpork 2010 project",
+    target_fixture='ridcully_member',
+)
+def when_accept_ridcully_member(
+    db_session,
+    client,
+    login,
+    ridcully_member,
+    user_vimes,
+    user_ridcully,
+    project_expo2010,
+    user_vetinari,
+) -> models.ProjectCrewMembership:
+    ridcully_member_accept = ridcully_member.accept(actor=user_ridcully)
+    db_session.add(ridcully_member_accept)
```

Comment:
`add` not required when calling `membership.accept` (same as `membership.replace`).